### PR TITLE
[#401] Input Bar and Write Messages

### LINKED ISSUE #401: Input bar + Write message
Create input bar and be able to write a message and show in the conversation.

### PR DESCRIPTION
**Pull request by ogbeche77**
Fixes #401

### CODE REVIEW

File: frontend/demo/src/actions/messages/index.ts [hunk 1]
```diff
@@ -2,13 +2,19 @@ import {Dispatch} from 'redux';
 import {createAction} from 'typesafe-actions';
 import {Message, ResponseMetadataPayload} from 'httpclient';
 import {HttpClientInstance} from '../../InitializeAiryApi';
+import {SendMessagesRequestPayload} from 'httpclient/payload/SendMessagesRequestPayload';
```
Comment (AitorAlgorta) [inline frontend/demo/src/actions/messages/index.ts:5]: make it one import `'httpclient'`

File: frontend/demo/src/actions/messages/index.ts [hunk 2]
```diff
@@ -30,3 +36,21 @@ export function listMessages(conversationId: string) {
       });
   };
 }
+export function sendMessages(conversationId: string, message: {text: string; type: string}) {
+  return async (dispatch: Dispatch<any>) => {
+    const requestPayload: SendMessagesRequestPayload = {conversation_id: conversationId, message};
```
Comment (AitorAlgorta) [inline frontend/demo/src/actions/messages/index.ts:41]: make the translation to `snake_case` inside the `httpclient`

File: frontend/demo/src/pages/Inbox/MessageInput/index.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+const mapDispatchToProps = {sendMessages};
+
+const connector = connect(null, mapDispatchToProps);
+type MessageInputProps = ConnectedProps<typeof connector>;
+
+const MessageInput = (props: MessageInputProps) => {
+  const [input, setInput] = useState('');
+  const textAreaAdjust = useRef(null);
+
+  const handleChange = event => {
```
Comment (AitorAlgorta) [inline frontend/demo/src/pages/Inbox/MessageInput/index.tsx:17]: missing type

File: frontend/demo/src/pages/Inbox/MessageInput/index.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        name="inputBar"
+        placeholder="Enter a message..."
+        autoFocus={true}
+        value={input}
+        onChange={handleChange}
+      />
+      <div className={styles.buttonSubmit}>
+        <button
+          type="button"
+          onClick={e => {
```
Comment (AitorAlgorta) [inline frontend/demo/src/pages/Inbox/MessageInput/index.tsx:44]: missing type

File: lib/typescript/httpclient/index.ts [hunk 5]
```diff
@@ -237,6 +239,15 @@ export class HttpClient {
       return error;
     }
   }
+
+  public async sendMessages(requestPayload: SendMessagesRequestPayload) {
+    try {
+      const response: MessagePayload = await this.doFetchFromBackend('messages.send', requestPayload);
+      return messageMapper(response);
+    } catch (error) {
```
Comment (AitorAlgorta) [inline lib/typescript/httpclient/index.ts:247]: type?

Comment (ogbeche77) [reply lib/typescript/httpclient/index.ts:247]: MessagePayload is a `type annotation` for the shape of the response. I had assume we needed it there.

**Review by bitboxer — COMMENTED** [review]
Looks really nice already. After the things that @AitorAlgorta mentioned this can be merged 🥳

**Review by bitboxer — DISMISSED** [review]
LGTM

File: lib/typescript/httpclient/index.ts (hunk 5 above)
Comment (AitorAlgorta) [reply lib/typescript/httpclient/index.ts:250]: I mean the Error @ogbeche77

**Review by thorstenairy — APPROVED** [review]
LGTM

**Review by bitboxer — DISMISSED** [review]
LGTM

**Pull request closed without merging**